crates/revm/src/evm_impl.rs

```diff
@@ -598,13 +601,30 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> EVMImpl<'a, GSPEC, DB,
 
     fn prepare_call(&mut self, inputs: &CallInputs) -> Result<PreparedCall, CallResult> {
         let gas = Gas::new(inputs.gas_limit);
-        // Load account and get code. Account is now hot.
-        let Some((bytecode, _)) = self.code(inputs.contract) else {
-            return Err(CallResult {
-                result: InstructionResult::FatalExternalError,
-                gas,
-                return_value: Bytes::new(),
-            });
+        let account = match self
+            .data
+            .journaled_state
+            .load_code(inputs.contract, self.data.db)
+        {
+            Ok((account, _)) => account,
+            Err(_) => {
+                return Err(CallResult {
+                    result: InstructionResult::FatalExternalError,
+                    gas,
+                    return_value: Bytes::new(),
+                })
+            }
+        };
+        let code_hash = account.info.code_hash();
+        let bytecode = match account.info.code {
+            Some(ref code) => code.clone(),
+            None => {
+                return Err(CallResult {
+                    result: InstructionResult::FatalExternalError,
+                    gas,
+                    return_value: Bytes::new(),
+                })
+            }
```

Comment:
```suggestion
        let bytecode = match account.info.code.unwrap_or_default();
```

code should be present, but to not use `expect` we can use `unwrap_or_default`

Reply:
Done in https://github.com/bluealloy/revm/pull/628/commits/6ba9372e6f3255155181175b7440401b8327adfc